src/build.ts

```diff
@@ -398,6 +416,7 @@ function validateLinks(outputs: Map<string, {resolvers: Resolvers}>): [valid: Li
 }
 
 function applyHash(path: string, hash: string): string {
+  if (path.startsWith("/_duckdb/")) return join("/_duckdb/", `${hash}-${path.slice("/_duckdb/".length)}`);
```

Comment:
First, why do we need special hashing behavior for DuckDB files? And second, if we do, we should have a separate function instead of putting this into `applyHash`. The only place we currently call `applyHash` with a path that starts with `/_duckdb/` is on L151, so we can just do the special behavior there. But I’d like to avoid the special behavior unless it’s necessary.

Reply:
We need this because when you `INSTALL foo FROM host` DuckDB will load it from `${host}/v1.1.1/wasm_{p}/parquet.duckdb_extension.wasm` and not accept anything else.

(If we could give the full path I would have gone with `${host}/v1.1.1/wasm_{p}/parquet.duckdb_extension.${hash}.wasm` but that doesn't seem to be supported. Maybe I missed something, though.)

Reply:
Yep makes sense once I grokked DuckDB’s requirements on the custom repositories for extensions. But I still think we should break out this into a separate function rather than overloading `applyHash`, and I’m not sure we need content hashes on anything under `_duckdb/` assuming that they don’t change the contents of extensions independently of the DuckDB version. If they do, then we’ll need a separate custom repository per extension since they’ll all have different hashes…

Reply:
Yes—currently I don't rely on any constancy from the extensions repository. Even if duckdb's hosted extensions are indeed changed only when the duckdb version changes, it won't necessarily be the case for custom extensions. Which is why each extension gets its own INSTALL FROM command with a specific hashed path.